### ISSUE TITLE
./bootstrap fails

### ISSUE THREAD

**Issue by rvanvelzen** [scfc/bison-php#14]
When executing  `./bootstrap` on Ubuntu 11.10, it fails due to `./build-aux/git-version-gen .tarball-version` resulting in `UNKNOWN`.

I'm not sure as to where this information should actually come from, but it seems a bit strange to me that it fails.

Any suggestions?

**Comment by scfc** [comment]
I don't know. I would start by checking out cleanly `upstream` which is bison's `master` (and reading README-hacking though you have probably done that :-)). If the error remains, the issue lies with bison, not with the PHP extension.

If all else fails, I can generate a tarball for you if you like.

**Comment by rvanvelzen** [comment]
Alright, what I've managed to figure out from trying to build bison's `master`, is that it is somewhat related to having `tags` in your git repo.

Look for the differences:
`richard@richard-pc:/mnt/data/dev/bison$ ./build-aux/git-version-gen .tarball-version
2.4.612-2a2b7`

And bison-php:
`richard@richard-pc:/mnt/data/dev/bison-php$ ./build-aux/git-version-gen .tarball-version
UNKNOWN`

Fortunately, a bit of hacking is never wrong, so I just touched a `.tarball-version` file until it worked... It has compiled and it works beautifully.

I'll be keeping an eye on this one. ;)

**Comment by lukescott** [comment]
This is still an issue. Merging with the upstream (currently) fixes it though. 

If anyone else comes across this:

Add "upstream" to your cloned repo if you haven't done so already:

git remote add upstream git://git.sv.gnu.org/bison.git

Then:

git fetch upstream
git merge upstream/master

I had to resolve a conflict with tests/local.mk. Simple enough to fix.

When compiling you have to fix src/flex-scanner.h by changing:

int  FLEX_PREFIX (get_leng) (void);

To:

yy_size_t  FLEX_PREFIX (get_leng) (void);

It _should_ compile after that. Since I was on a Mac and my autoconf was out of date I downloaded the latest versions of autoconf, automake, gettext, and perhaps a couple other gnu libraries and installed them with the prefix of "/gnu". Then I had to do "export PATH=/gnu:$PATH" so the new ones would be used.

Hopefully this helps someone!

**Comment by scfc** [comment]
I must admit that I pulled from upstream, merged the conflicts and then pushed to GitHub aware of the errors without considering that someone will use master as "always working" :-).  Akim fixed the issues in upstream since, and I merged and pushed them here as well.

So, not to play the blame game, but the master branch in bison-php will usually be Bison's master branch plus PHP patches.  I hope we will integrate bison-php soon in upstream, so that this will be clearer and that to use bison-php you won't have to jump through all the same hoops as a Bison developer, but can just use a regular release.
